cpp/src/Ice/ConnectionI.cpp

```diff
@@ -1279,21 +1279,16 @@ Ice::ConnectionI::getEndpoint() const noexcept
     return _endpoint; // No mutex protection necessary, _endpoint is immutable.
 }
 
-optional<ObjectPrx>
+ObjectPrx
 Ice::ConnectionI::createProxy(const Identity& ident) const
 {
+    // TODO: merge my other PR code
+
     // Create a reference and return a reverse proxy for this reference.
     ReferencePtr ref =
         _instance->referenceFactory()->create(ident, const_cast<ConnectionI*>(this)->shared_from_this());
 
```

Comment:
Should we assert "ref" is not null? does the reference factory ever return `null` from create?

Reply:
That's a good question. And no, ref can never but null, and I added a few asserts in ReferenceFactory to make sure the identity's name it gets is never empty.